src/Microsoft.AspNetCore.Authentication.JwtBearer/JwtBearerOptions.cs

```diff
@@ -118,5 +118,21 @@ public JwtBearerOptions() : base()
         /// <see cref="Http.Authentication.AuthenticationProperties"/> after a successful authorization.
         /// </summary>
         public bool SaveToken { get; set; } = true;
+
+        /// <summary>
+        /// Authentication failures with one of these exception types, or types derived from these, will have their messages
+        /// included in a 'X-Authentication-Failure' response header.
+        /// </summary>
+        public ICollection<Type> ReportableFailures { get; set; } = new List<Type>()
```

Comment:
Is it really worth providing an option for that? One could use the `Challenge` event to replace the `error`/`error_description` in the `WWW-Authenticate` header.